packages/story-editor/src/components/colorPicker/karma/colorPicker.karma.js

```diff
@@ -158,7 +158,12 @@ describe('ColorPicker', () => {
 
         // Add text and apply the previously saved color.
         await fixture.events.click(fixture.editor.library.textAdd);
-        await waitFor(() => fixture.editor.canvas.framesLayer.frames[1].node);
+        await waitFor(() => {
+          if (!fixture.editor.canvas.framesLayer.frames[1].node) {
+            throw new Error('node not ready');
+          }
+          expect(fixture.editor.canvas.framesLayer.frames[1].node).toBeTruthy();
```

Comment:
if this expect was only ever in here for the `waitFor` statement, I wonder if we even need it anymore?

Reply:
Above we are clicking the `addText` button. Since these test can run kinda fast in ci, the node hasn't always been mounted to the dom by the time we need to interact with it. So we added the `waitFor` to force karma to repoll until the node appears on the dom. At least that is my understanding. Otherwise we end up with a flakey test that doesn't always find the node it is looking for.